cts.json

```
@@ -4096,6 +4153,22 @@
       ],
       "result": []
     },
+    {
+      "name": "functions, length, filter on primitive array element, selects nothing",
+      "selector": "$[?length(@)==1]",
```

This test could be made broader, thus:
```suggestion
      "selector": "$[?length(@)>=0]",
```

---

Also, it would be good to add a test for the comparison of two `Nothing` values if we don't already, e.g.:
```
$[?length(@)==length(@)]
```
should select all the elements of an array argument.

---

https://github.com/jsonpath-standard/jsonpath-compliance-test-suite/blob/main/tests%2Ffilter.json

We have `Nothing` equality.

---

@glyn I think these changes are moot given the conversation I've raised.  What these tests are trying to verify can be accomplished with several more targeted tests.